src/js/services/app.js

```diff
@@ -74,24 +75,41 @@ export function showChannelPage(channelType) {
         const {user, app: {integrations}} = getState();
         const channelDetails = CHANNEL_DETAILS[channelType];
         const isLinked = isChannelLinked(user.clients, channelType);
-        const openLink = channelDetails.getURL && (!channelDetails.Component || isLinked);
 
-        if (openLink) {
+        function _showChannelPage() {
+            dispatch(AppStateActions.showChannelPage(channelType));
+
+            return dispatch(connectToFayeUser())
+                .then(() => dispatch(channelDetails.onChannelPage()));
+        }
+
+        function openLink() {
             const appChannel = getIntegration(integrations, channelType);
-            const link = channelDetails.getURL(user, appChannel, isLinked);
+            const getCode = channelDetails.usesCode ?
```

Comment:
the popup blocker isn't triggered if `window.open` is called in a direct response to a user action... I think once you add a promise in the mix, you exit the current event stack and the browser doesn't consider it in response to a user action and triggers the popup blocker.